fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ReviewQualityCheckBottomSheet.kt

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.shopping.ui
+
+import androidx.compose.animation.animateContentSize
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import mozilla.components.lib.state.ext.observeAsState
+import org.mozilla.fenix.shopping.state.ReviewQualityCheckAction
+import org.mozilla.fenix.shopping.state.ReviewQualityCheckState
+import org.mozilla.fenix.shopping.state.ReviewQualityCheckState.OptedIn.ProductReviewState.ProductAnalysis
+import org.mozilla.fenix.shopping.state.ReviewQualityCheckStore
+
+/**
+ * Top-level UI for the Review Quality Check feature.
+ *
+ * @param store [ReviewQualityCheckStore] that holds the state.
+ * @param onRequestDismiss Invoked when a user action requests dismissal of the bottom sheet.
+ * @param modifier The modifier to be applied to the Composable.
+ */
+@Composable
+fun ReviewQualityCheckBottomSheet(
```

Comment:
could we get a preview or two stood up? it might help to emulate `TabsTrayPreviewRoot` and have a series of previews all derive from one preview composable. that way all of the business logic, remember variables, and store instantiation is in one location.

Reply:
Sure thing, I'd like to do that in a follow up PR as the feature develops. Currently the only state that's available is `ProductAnalysis` and the preview for that is there.